server/src-lib/Hasura/Tracing.hs

```diff
@@ -197,42 +202,64 @@ instance MonadTrace m => MonadTrace (ExceptT e m) where
   currentReporter = lift currentReporter
   attachMetadata = lift . attachMetadata
 
+-- | Create a Random 16 character hex string
+randomHex :: IO BS.ByteString
+randomHex = do
+  randNum <- liftIO (Rand.randomIO :: IO Word64)
+  let hexString = Hex.encode (BL.toStrict (BIN.encode  randNum))
+  return hexString
+
 -- | Inject the trace context as a set of HTTP headers.
 injectHttpContext :: TraceContext -> [HTTP.Header]
 injectHttpContext TraceContext{..} = 
-  [ ("X-Hasura-TraceId", fromString (show tcCurrentTrace))
-  , ("X-Hasura-SpanId", fromString (show tcCurrentSpan))
-  ]
+  case tcCurrentParent of
+    Nothing -> [ ("X-B3-TraceId", fromString (show tcCurrentTrace))
```

Comment:
You don't want to use `show` here or you'll end up with quotes around the strings.